### DISCUSSION TITLE
Help with SGBA algorithm.

### DISCUSSION THREAD

**Discussion by Fygas1** [bitcraze/discussions#1602] [Q&A]
Hey I'm new to this and what caught my attention is this algorithm of swarm exploration and I'm now looking to run it (https://github.com/tudelft/SGBA_code_SR_2019)

I tried to follow the steps but i'm unable to run it. After running the flash_all_crazyflie.sh script (step 2), i get the following and nothing happens. https://imgur.com/d5aNPpm

I'm using Crazyflie 2.1 and I get that this is an old firmware. It says the algorithm is being updated for Crazyflie 2.1 firmware and directs me to [SGBA_CF2_App_layer](https://github.com/tudelft/SGBA_CF2_App_layer), which mentions that it’s not the original source code used for experiments. However, the original repo also directs back to the new one.

Which repo should I use to run the actual algorithm? Is the first original repo just not working for Crazyflie 2.1? If you could guide me on where to start off I would really appreciate it. Thank you so much.

**Comment by gemenerik** [comment]
Hi Andreas, 

If you want to use it on the CF2.1 and the original repository supports only the CF2.0, I would follow their advice to use the CF2.1-compatible repository, otherwise it will likely not work.

However, since I’m not familiar with this repository and it’s maintained by TU Delft, I recommend asking any further questions directly there. They’ll be able to provide the best support.
